pl.c

```diff
@@ -641,14 +646,24 @@ struct track_info *pl_play_selected_row(void)
 	 */
 
 	int was_in_track_window = pl_cursor_in_track_window;
-	if (!pl_cursor_in_track_window)
+
+	struct track_info *rv = NULL;
+
+	if (!pl_cursor_in_track_window) {
+		if (shuffle && !pl_empty(pl_visible)) {
+			struct shuffle_track *st = shuffle_list_get_next(&pl_visible->shuffle_root, NULL, pl_dummy_filter);
+			struct simple_track *track = &st->simple_track;
+			rv = pl_play_track(pl_visible, track);
```

Comment:
Am I right that `window_goto_top` below will select the first track, even though we may not be playing it?

Reply:
Yes, that's right.

BTW, you can select more lines by dragging the comment `+`.

Reply:
As far as I understand it, the unconditional `struct track_info *rv = pl_play_selected_track();` was the problem. I'm not sure what the `window_goto_top` is here for.

Reply:
I suppose `window_goto_top` only makes sense when not shuffling. Instead, it would be better to goto the position of the current track. I will see if I can add that.

Reply:
OK, I've pushed an update for that